web/source/kmwcallback.ts

```diff
@@ -628,23 +659,112 @@ class KeyboardInterface {
    * @return      {boolean}           True if character found in 'any' string, sets index accordingly
    * Description  Test for character matching
    */    
-  any(n: number, ch: string, s:string): boolean {
+  any(n: number, ch: KeyboardStoreElement, s: KeyboardStore): boolean {
     if(ch == '') {
       return false;
     }
-    var Lix = s._kmwIndexOf(ch); //I3319
+    
+    s = this._ExplodeStore(s);
+    var Lix = -1;
+    for(var i=0; i < s.length; i++) {
+      if(typeof(s[i]) == 'string') {
+        if(s[i] == ch) {
+          Lix = i;
+          break;
+        }
+      } else if(s[i]['d'] === ch['d']) {
+        Lix = i;
+        break;
+      }
+    }
     this._AnyIndices[n] = Lix;
     return Lix >= 0;
   }
+
+  /**
+   * Function     _Index
+   * Scope        Public 
+   * @param       {string}      Ps      string
+   * @param       {number}      Pn      index
+   * Description  Returns the character from a store string according to the offset in the index array
+   */
+  _Index(Ps: KeyboardStore, Pn: number): KeyboardStoreElement {        
+    Ps = this._ExplodeStore(Ps);
+
+    if(this._AnyIndices[Pn-1] < Ps.length) {   //I3319
+      return Ps[this._AnyIndices[Pn-1]];
+    } else {
+      /* Should this really be possible for a compiled keyboard?  
+       * Should we throw an error / output a console.error("")?
+       */
+      return "";
+    }
+  }
+
+  /**
+   * Function     indexOutput   KIO
+   * Scope        Public
+   * @param       {number}      Pdn     no of character to overwrite (delete)
+   * @param       {string}      Ps      string
+   * @param       {number}      Pn      index
+   * @param       {Object}      Pelem   element to output to
+   * Description  Output a character selected from the string according to the offset in the index array
+   */
+  indexOutput(Pdn: number, Ps: KeyboardStore, Pn: number, Pelem: HTMLElement): void {
+    this.resetContextCache();
+
+    var indexChar = this._Index(Ps, Pn);
+    if(typeof(indexChar) == 'string' ) {
+      this.output(Pdn,Pelem,indexChar);  //I3319
+    } else {
+      this.deadkeyOutput(Pdn, Pelem, indexChar['d']);
+    }
+  }
+  
   
+  /**
+   * Function     deleteContext KDC  
+   * Scope        Public
+   * @param       {number}      dn      number of context entries to overwrite
+   * @param       {Object}      Pelem   element to output to 
+   * @param       {string}      s       string to output   
+   * Description  Keyboard output
+   */
+  deleteContext(dn: number, Pelem): void {
+    var context: CachedExEntry;
+
+    // We want to control exactly which deadkeys get removed.
+    if(dn > 0) {
```

Comment:
Do you want to range-check the maximum number of context entries allowed to overwrite?

Reply:
I don't think we need to do that here. It doesn't add any safety and is unlikely to arise.